cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -464,7 +465,6 @@ struct DivideChecked {
   template <typename T, typename Arg0, typename Arg1>
   static enable_if_integer_value<T> Call(KernelContext*, Arg0 left, Arg1 right,
                                          Status* st) {
-    static_assert(std::is_same<T, Arg0>::value && std::is_same<T, Arg1>::value, "");
```

Comment:
For which reason was this failing? Shouldn't it receive `int64_t` for all arguments in this PR?

Reply:
I assumed it would but does not. Returning.